main.py

```diff
@@ -101,11 +84,64 @@ def process_data():
     print(df4.to_string())
     df4.to_csv("filtered_data.csv")
 
```

Comment:
because in python you can pass anything and return anything that makes the code hard to maintain
so python introduced "annotations"
https://www.geeksforgeeks.org/function-annotations-python/

so for this function it would look like:
def cal_dividend_increament(div_obj: pandas.core.series.Series, p:int) -> tuple:

annotations are not a must, but the IDE enforces them and it makes the code a lot more robust